pkg/controller/restore-operator/http.go

```diff
@@ -39,27 +40,42 @@ func (r *Restore) startHTTP() {
 	panic(http.ListenAndServe(listenAddr, nil))
 }
 
-// serveBackup parses incoming request url of the form /backup/<cluster-name>
-// get the etcd cluster name.
+// serveBackup parses incoming request url of the form /backup/<restore-name>
+// get the etcd restore name.
 // Then it returns the etcd cluster backup snapshot to the caller.
 func (r *Restore) serveBackup(w http.ResponseWriter, req *http.Request) {
-	clusterName := string(req.URL.Path[len(backupHTTPPath):])
-	if len(clusterName) == 0 {
-		http.Error(w, "cluster is not specified", http.StatusBadRequest)
+	restoreName := string(req.URL.Path[len(backupHTTPPath):])
+	if len(restoreName) == 0 {
+		http.Error(w, "restore name is not specified", http.StatusBadRequest)
```

Comment:
Should have been: http.StatusBadRequest -> http.StatusNotFound

Can you change it alongside?